src/overlap_pmmg.c

```diff
@@ -0,0 +1,736 @@
+    PMMG_CALLOC(parmesh,pointTagPBDY_ToSend,         mesh->np, uint16_t,"pointTagPBDY_ToSend",      ier = 0);
+    PMMG_CALLOC(parmesh,pointRefInterior_ToSend,     mesh->np, int,     "pointRefInterior_ToSend",  ier = 0);
+    PMMG_CALLOC(parmesh,pointRefPBDY_ToSend,         mesh->np, int,     "pointRefPBDY_ToSend",      ier = 0);
+    PMMG_CALLOC(parmesh,pointIdxPBDY_ToSend,         mesh->np, int,     "pointIdxPBDY_ToSend",      ier = 0);
+    PMMG_CALLOC(parmesh,pointIdxInterface_ToSend,    nitem_ext,int,     "pointIdxInterface_ToSend", ier = 0);
+    PMMG_CALLOC(parmesh,tetraVertices_ToSend,      4*mesh->ne, int,     "tetraVertices_ToSend",     ier = 0);
+    PMMG_CALLOC(parmesh,tetraVerticesSeen_ToSend,  4*mesh->ne, int,     "tetraVerticesSeen_ToSend", ier = 0);
+    PMMG_CALLOC(parmesh,tetraRef_ToSend,             mesh->ne, int,     "tetraRef_ToSend",          ier = 0);
+    PMMG_CALLOC(parmesh,lsInterior_ToSend,           mesh->np, double,  "lsInterior_ToSend",        ier = 0);
+    PMMG_CALLOC(parmesh,lsPBDY_ToSend,               mesh->np, double,  "lsPBDY_ToSend",            ier = 0);
+
+    /** STEP 1 - Identify nodes at the interface between Pcolor_in and Pcolor_out
+              a) Assign flag -1 to these nodes
+              b) Store the local point indices in pointIdxInterface_ToSend
+                 to be able to fill hash_out2in **/
+    /* Loop over the nodes in the external node communicator */
+    for (i=0; i < nitem_ext; i++) {
+      /* Get the index of the node in internal communicator */
+      idx = ext_comm->int_comm_index[i];
+      ip  = int_comm->intvalues[idx];
+
+      /* Add the flag -1 to this point */
+      p0 = &mesh->point[ip];
+      p0->flag = -1;
+      pointIdxInterface_ToSend[i] = ip;
+    }
+
+    /** STEP 2 - Identification of points and tetras to send to Pcolor_out
+         tetraVerticesSeen_ToSend: flag the tetra vertices using following rules
+           +1 means the vertex is seen for the first time
+            0 means the vertex has been already seen (from another tetra)
+              OR is on the interface between Pcolor_in and Pcolor_out
+              OR is tagged MG_PARBDY (special treatments for those) **/
+    /* Loop over the tetra on this partition, i.e. Pcolor_in */
+    for (k=1; k<=nt_initial; k++) {
+      pt  = &mesh->tetra[k];
+
+      /* If one vertex if flag -1, assign MG_OVERLAP to this tetra */
+      for (i=0; i<4; i++) {
+        ip = pt->v[i];
+        p0 = &mesh->point[ip];
+        if ( p0->flag < 0 ) {
+          pt->tag |= MG_OVERLAP;
+          break;
+        }
+      }
+
+      /* If tetra has not been identified as MG_OVERLAP, then ignore it */
+      if ( !(pt->tag & MG_OVERLAP) ) continue;
+
+      tetraRef_ToSend[ntTot_in2out] = pt->ref;
+
+      for (i=0; i<4; i++) {
+        ip  = pt->v[i];
+        p0  = &mesh->point[ip];
+        tetraVertices_ToSend[4*ntTot_in2out+i] = ip;
+
+        /* If this node has never been seen before (p0->flag!=color_out+1)
```

Comment:
Yes exactly. Comments have been modified as follows:
- Before the loop on the 4 vertices of the overlap tetra, comment has been added `/* Loop over the vertices of this tetra, all the nodes belongs to the overlap
         that we want to send to Pcolor_out */`
- The comment before `if ( (p0->flag>=0)  && (p0->flag!=color_out+1) )` has been improved and modified as follows: `        /* If this node has never been seen before for the partition on Pcolor_out
           (i.e. p0->flag!=color_out+1) and is not on interface between Pcolor_in
           and Pcolor_out (p0->flag>=0):
            a) in VerticesSeen_ToSend : assign 1 if this node is not tagged MG_PARBDY;
               assign 0 otherwise (special treatment for those).
            b) store its coordinates in pointCoord*_ToSend, its tag in
               pointTag*_ToSend; its ref in pointRef*_ToSend; its level-set value
               in ls*_ToSend. `*` being either `Interior` for pts not tagged
               MG_PARBDY or `PBDY` for pts tagged MG_PARBDY */

Does it sound clearer now? Or still need improvements? Thanks.